test/framework/deployment_helpers.go

```diff
@@ -179,51 +186,117 @@ type watchPodLogsInput struct {
 // retrieving the deployment, its pods, or setting up a log file. If there is an error with the log streaming itself,
 // that does not cause the test to fail.
 func watchPodLogs(ctx context.Context, input watchPodLogsInput) {
-	pods := &corev1.PodList{}
-	Expect(input.GetLister.List(ctx, pods, client.InNamespace(input.Namespace), client.MatchingLabels(input.Labels))).To(Succeed(), "Failed to list Pods for deployment %s", klog.KRef(input.Namespace, input.DeploymentName))
+	// Create informer to watch for pods matching input.
 
-	for _, pod := range pods.Items {
-		for _, container := range pod.Spec.Containers {
-			log.Logf("Creating log watcher for controller %s, pod %s, container %s", klog.KRef(input.Namespace, input.DeploymentName), pod.Name, container.Name)
+	podInformer, err := input.Cache.GetInformer(ctx, &corev1.Pod{})
+	Expect(err).ToNot(HaveOccurred(), "Failed to create controller-runtime informer from cache")
 
-			// Create log metadata file.
-			logMetadataFile := filepath.Clean(path.Join(input.LogPath, input.DeploymentName, pod.Name, container.Name+"-log-metadata.json"))
-			Expect(os.MkdirAll(filepath.Dir(logMetadataFile), 0750)).To(Succeed())
+	selector, err := metav1.LabelSelectorAsSelector(input.LabelSelector)
+	Expect(err).NotTo(HaveOccurred())
 
-			metadata := logMetadata{
-				Job:       input.Namespace + "/" + input.DeploymentName,
-				Namespace: input.Namespace,
-				App:       input.DeploymentName,
-				Pod:       pod.Name,
-				Container: container.Name,
-				NodeName:  pod.Spec.NodeName,
-				Stream:    "stderr",
+	handlerRegistration, err := podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
```

Comment:
Added, PTAL :-)

context handling is not great IMHO, but I think its better to pass the single context down, so finishing the context would lead to stopping all log streamings.

Reply:
Looks fine. No alternatives I think and the behavior isn't really different than before the change